src/dmd/frontend.d

```diff
@@ -54,3 +62,161 @@ void initDMD()
     builtin_init();
 }
 
+/**
+Add import paths to the `global.path`.
+Params:
+    path = range of imports to add
+*/
+void addImports(T)(T path)
+{
+    import dmd.globals : global;
+    import dmd.arraytypes : Strings;
+
+    Strings* res = new Strings();
+    foreach (p; path)
+    {
+        import std.string : toStringz;
+        Strings* a = new Strings();
+        a.push(p.toStringz);
+        res.append(a);
+    }
+    global.path = res;
+}
+
+/**
+Finds a dmd.conf and parses it for import paths.
+This depends on the `$DMD` environment variable.
+If it's set to `ldmd`, it will try to detect and parse an `ldc2.conf` instead.
+
+Returns:
+    A forward range of normalized import paths.
+*/
+auto findImportPaths()
+{
+    import std.file : exists, getcwd;
+    import std.string : fromStringz, toStringz;
+    import std.path : buildPath, buildNormalizedPath, dirName;
+    import std.process : env = environment, execute;
+    import dmd.dinifile : findConfFile;
+    import dmd.errors : fatal;
+    import std.algorithm, std.range, std.regex;
+    import std.stdio : File,  stderr;
+
+    auto dmdEnv = env.get("DMD", "dmd");
+    // TODO: use a more fault-proof detection
+    auto whichDMD = execute(["which", dmdEnv]);
+    if (whichDMD.status != 0)
+    {
+        stderr.writeln("Can't find DMD.");
+        fatal;
+    }
+
+    immutable dmdFilePath = whichDMD.output;
+    string iniFile;
+
+    if (dmdEnv.canFind("ldmd"))
+    {
+        immutable ldcConfig = "ldc2.conf";
+        immutable binDir = dmdFilePath.dirName;
+        // https://wiki.dlang.org/Using_LDC
+        auto ldcConfigs = [
+            getcwd.buildPath(ldcConfig),
+            binDir.buildPath(ldcConfig),
+            binDir.dirName.buildPath("etc", ldcConfig),
+            "~/.ldc".buildPath(ldcConfig),
+            binDir.buildPath("etc", ldcConfig),
+            binDir.buildPath("etc", "ldc", ldcConfig),
+            "/etc".buildPath(ldcConfig),
+            "/etc/ldc".buildPath(ldcConfig),
+        ].filter!exists;
+        assert(!ldcConfigs.empty, "No ldc2.conf found");
+        iniFile = ldcConfigs.front;
+    }
+    else
+    {
+        auto f = findConfFile(dmdFilePath.toStringz, "dmd.conf");
+        iniFile = f.fromStringz.idup;
+        assert(iniFile.exists, "No dmd.conf found.");
+    }
+
+    return File(iniFile, "r")
+        .byLineCopy
+        .map!(l => l.matchAll(`-I[^ "]+`.regex)
+                    .joiner
+                    .map!(a => a.drop(2)
+                                .replace("%@P%", dmdFilePath.dirName)
+                                .replace("%%ldcbinarypath%%", dmdFilePath.dirName)))
+        .joiner
+        .array
+        .sort
+        .uniq
+        .map!buildNormalizedPath;
```

Comment:
This whole expression doesn't look very nice. Perhaps split it up with some helper functions/aliases. It's not very clear of what the code is actually doing.